Sources/WinAppDriver/WinAppDriver.swift

```diff
@@ -34,12 +40,59 @@ public class WinAppDriver: WebDriver {
         executablePath: String = defaultExecutablePath,
         ip: String = defaultIp,
         port: Int = defaultPort,
-        waitTime: TimeInterval? = defaultStartWaitTime) throws -> WinAppDriver {
-
+        waitTime: TimeInterval? = defaultStartWaitTime,
+        outputFile: String? = nil) throws -> WinAppDriver {
         let processTree: Win32ProcessTree
+        var childStdinHandle: HANDLE? = nil
         do {
-            processTree = try Win32ProcessTree(path: executablePath, args: [ ip, String(port) ])
+            var launchOptions = ProcessLaunchOptions()
+            if let outputFile = outputFile {
+                // Open the output file for writing to the child stdout.
+                var securityAttributes = SECURITY_ATTRIBUTES()
+                securityAttributes.nLength = DWORD(MemoryLayout<SECURITY_ATTRIBUTES>.size)
+                securityAttributes.bInheritHandle = true
+                launchOptions.stdoutHandle = try outputFile.withCString(encodedAs: UTF16.self) {
+                    outputFile throws in
+                    CreateFileW(
+                        UnsafeMutablePointer<WCHAR>(mutating: outputFile), DWORD(GENERIC_WRITE),
+                        DWORD(FILE_SHARE_READ), &securityAttributes,
+                        DWORD(OPEN_ALWAYS), DWORD(FILE_ATTRIBUTE_NORMAL), nil)
+                }
+                if launchOptions.stdoutHandle == INVALID_HANDLE_VALUE {
+                    // Failed to open the output file for writing.
+                    throw Win32Error.getLastError(apiName: "CreateFileW")
+                }
+
+                // Use the same handle for stderr.
+                launchOptions.stderrHandle = launchOptions.stdoutHandle
+
+                // WinAppDriver will close immediately if no stdin is provided so create a dummy
+                // pipe here to keep stdin open until the child process is closed.
+                var childReadInputHandle: HANDLE?
+                if !CreatePipe(&childReadInputHandle, &childStdinHandle, &securityAttributes, 0) {
+                    CloseHandle(launchOptions.stdoutHandle)
```

Comment:
Better move the defer block before the if on line 49 than duplicate the CloseHandle.

Reply:
I meant to move it before line 49, outside of the if block.

Reply:
Ah I see, I'll follow-up!